xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -14956,11 +14956,17 @@ glusterd_add_peers_to_auth_list(char *volname)
                "auth allow list is not set");
         goto out;
     }
+
+    /* Here calculates length of the new buffer.
+     * It is prepared for the worst-case scenario where every peeers
+     * were not in the original auth.allow list and needed to be added
+     */
+    len = strlen(auth_allow_list);
```

Comment:
You need to add 1 also to avoid string overflow otherwise buffer is corrupted at the time of of strncat due to this test case (subdir-mount.t) is crashed.

Reply:
Thanks for pointing out. Obviously I have forgot about the NUL character. I have amended and re-pushed the commit, along with a new commit message.